internal/recert/recert.go

```diff
@@ -20,43 +20,50 @@ const (
 var staticDirs = []string{"/kubelet", "/kubernetes", "/machine-config-daemon"}
 
 type RecertConfig struct {
-	DryRun            bool     `json:"dry_run,omitempty"`
-	ExtendExpiration  bool     `json:"extend_expiration,omitempty"`
-	ForceExpire       bool     `json:"force_expire,omitempty"`
-	EtcdEndpoint      string   `json:"etcd_endpoint,omitempty"`
-	ClusterRename     string   `json:"cluster_rename,omitempty"`
-	SummaryFile       string   `json:"summary_file,omitempty"`
-	SummaryFileClean  string   `json:"summary_file_clean,omitempty"`
-	StaticDirs        []string `json:"static_dirs,omitempty"`
-	StaticFiles       []string `json:"static_files,omitempty"`
-	CNSanReplaceRules []string `json:"cn_san_replace_rules,omitempty"`
-	UseKeyRules       []string `json:"use_key_rules,omitempty"`
-	UseCertRules      []string `json:"use_cert_rules,omitempty"`
+	DryRun           bool   `json:"dry_run,omitempty"`
+	ExtendExpiration bool   `json:"extend_expiration,omitempty"`
+	ForceExpire      bool   `json:"force_expire,omitempty"`
+	EtcdEndpoint     string `json:"etcd_endpoint,omitempty"`
+	ClusterRename    string `json:"cluster_rename,omitempty"`
+	Hostname         string `json:"hostname,omitempty"`
+	// We intentionally don't omitEmpty this field because an empty string here
+	// means "delete the kubeadmin password secret" while a complete omission
+	// of the field means "don't touch the secret". We never want the latter,
+	// we either want to delete the secret or update it, never leave it as is.
+	KubeadminPasswordHash string   `json:"kubeadmin_password_hash"`
```

Comment:
If `don't touch the secret` isn't a valid option, you can interpret a complete omission of the filed as `delete the kubeadmin secret`.

Reply:
Recert runs in more contexts other than just seed reconfiguration (e.g. seed expiration)

In those contexts, I don't want recert to do anything with this secret.

So recert supports:

- Don't touch the secret (omission of field)
- Modify the secret (field set to non-empty value)
- Delete the secret (field set to empty value)